docs/ops/doc/IntrospectingOps.md

```diff
@@ -0,0 +1,124 @@
+# Reasoning about Ops
+
+The declarative nature of Ops is convenient for matching and execution, but sometimes it is necessary to know where an Op comes from. This page describes the tools provided by SciJava Ops to understand the code underneath.
+
+Naturally, to understand where an Op came from, you first need an Op:
+
+```java
+var img = ...;
+var out = ...;
+var sigma = 5.0;
+
+// This Op call finds a "filter.gauss Op" that blurs an image using a provided
+// sigma, placing the result in the output container "out".
+//
+// NB the Op is saved within the variable "op"
+var op = ops.op("filter.gauss").input(img, sigma).output(out).computer();
+```
+
+## Introduction
+
+Each Op is generated from an `OpInfo` object, describing an Op's source code, version, and other associated metadata. If you've matched an Op `op`, you can obtain its `OpInfo` using the method `Ops.info`:
+
+```java
+import org.scijava.ops.api.Ops;
+
+var info = Ops.info(op);
+```
+
+The `OpInfo` often provides all needed information, but if `op` utilizes dependencies or other framework features, you may wish to introspect the "rich" Op instead, using the method `Ops.rich`.
+
+```java
+var richOp = Ops.rich(op);
+```
+
+These two methods provide gateways into learning more about where your Op came from!
+
+## Implementations and Versions
+
+For reproducibility, it is important to know not only the underlying implementations, or discrete lines of code, backing each executed Op, but additionally the *version* of that implementation if it changes over time. Fortunately, SciJava Ops provides mechanisms to determine both!
+
+A unique identifier to an Op's implementation can be obtained from its `OpInfo` using the method `OpInfo.implementationName()`. For example, the following
+
+```java
+System.out.println(Ops.info(op).implementationName());
+```
+
+might print out (if you're using `scijava-ops-image` version `1.0.0`):
+
+```
+org.scijava.ops.image.filter.gauss.Gaussians.gaussRAISingleSigma(net.imglib2.RandomAccessibleInterval<I>,double,net.imglib2.outofbounds.OutOfBoundsFactory<I, net.imglib2.RandomAccessibleInterval<I>>,net.imglib2.RandomAccessibleInterval<O>)Reduction1
+```
+
+Note that in many cases, including the case above, the implementation name may not itself be an explicit piece of code, however it must always explain how an explicit piece of code became the Op you've matched.
+
+Similarly, the method `OpInfo.version()` describes **the version of the Op**, which is set by whichever component declares the Op. This version might change due to:
+* An update in the version of the underlying implementation (such as a bump in the version of `scijava-ops-image`)
```

Comment:
Does this mean that a bump in the version of OpenCV could affect the Op versions? Based on our previous discussion today, I thought the answer was no...

Reply:
If the `ops.yaml` op versions got pulled from the version of the class housing the static method being wrapped... then the Op version *would* correspond to the version of OpenCV/JavaCV, which would be nice.

Reply:
Well, I say *might*, because in the case where a Java version of a dependency is pinned, like `org.bytedeco.opencv`, or the `scijava-ops-image` version, then a bump in the version would affect the Op version, as we'd have to update the Op library to take advantage of that.

I don't know enough about the `OpenCV` C++ code update, but I think the *might* gives us some wiggle room. I think this also holds us to a good standard which we should try to live up to in the future.